### PR TITLE
Remove ClientContext

### PR DESCRIPTION
**Pull request by mfedderly**
This was an unnecessary wrapper around calls to `convertWireToOsdkObject` and `ThinClient`

### CODE REVIEW

File: packages/legacy-client/src/client/convertWireToOsdkObject.ts [hunk 1]
```diff
@@ -87,29 +91,30 @@ function createPrototype<
...
 export function convertWireToOsdkObject<
-  T extends keyof O["objects"] & string,
+  T extends ObjectTypesFrom<O> & string,
   O extends OntologyDefinition<any>,
 >(
-  context: ClientContext,
-  obj: OsdkLegacyPropertiesFrom<O, T> & OntologyObject<T>,
+  client: ThinClient<O>,
+  apiName: T,
+  obj: OntologyObjectV2,
```
Comment (mfedderly) [inline packages/legacy-client/src/client/convertWireToOsdkObject.ts:99]: This change right here is pretty important. It effectively gives us `Record<string, any>`, and we're depending on it to contain `__rid` and `__primaryKey` on it.

File: packages/legacy-client/src/client/convertWireToOsdkObject.ts [hunk 2]
```diff
@@ -87,29 +91,30 @@ function createPrototype<
...
 ): OsdkLegacyObjectFrom<O, T> {
-  const ontologyCache = cache.get(context.ontology.metadata.ontologyRid);
-  let proto = ontologyCache?.get(obj.__apiName);
+  const ontologyCache = cache.get(client.ontology.metadata.ontologyRid);
+  let proto = ontologyCache?.get(apiName);
 
   if (proto == null) {
-    let proto = createPrototype(
-      context,
+    proto = createPrototype(
```
Comment (mfedderly) [inline packages/legacy-client/src/client/convertWireToOsdkObject.ts:105]: 🤦 shadowed my proto variable

File: packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/getObject.ts [hunk 3]
```diff
@@ -14,32 +14,33 @@
...
 import type { OntologyObject } from "../../baseTypes";
 import { GetObjectErrorHandler, handleGetObjectError } from "../ErrorHandlers";
 import type { GetObjectError } from "../Errors";
 import type { Result } from "../Result";
-import type { ClientContext } from "./ClientContext";
 import { wrapResult } from "./util/wrapResult";
 
 export async function getObject<T extends OntologyObject>(
-  context: ClientContext,
+  client: ThinClient<any>,
```
Comment (pashakondratyev) [inline packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/getObject.ts:28]: I can take care of this in a follow up but we can probably change all the calls to take the OntologyDefinition as a type and use that here

File: packages/legacy-client/src/client/convertWireToOsdkObject.ts (hunk 1 above)
Comment (pashakondratyev) [reply packages/legacy-client/src/client/convertWireToOsdkObject.ts:99]: Yeah unfortunately we need to make that assumption at some point

**Review by pashakondratyev — APPROVED** [review]

**Pull request merged**